File: backend/src/internal/models/seller.go
/ backend/src/internal/models/seller.go
package models

// Seller represents a marketplace seller
type Seller struct {
	ID             string           `json:"id"`
	Nickname       string           `json:"nickname"`
	Reputation     SellerReputation `json:"reputation"`
	Tags           []string         `json:"tags"`
	Location       Location         `json:"location"`
	ExperienceYears int             `json:"experience_years"`
	LeaderStatus   string           `json:"leader_status"`
}

// SellerReputation contains seller reputation metrics
type SellerReputation struct {
	Level        int                    `json:"level"`
	Status       string                 `json:"status"` // platinum, gold, silver
	Transactions TransactionMetrics     `json:"transactions"`
	Ratings      RatingMetrics          `json:"ratings"`
}

// TransactionMetrics for seller transactions
type TransactionMetrics struct {
	Completed int `json:"completed"`
	Canceled  int `json:"canceled"`
	Total     int `json:"total"`
}

// RatingMetrics for seller ratings
type RatingMetrics struct {
	Positive float64 `json:"positive"`
	Neutral  float64 `json:"neutral"`
	Negative float64 `json:"negative"`
}

// Location represents geographical location
type Location struct {
	City    string `json:"city"`
	State   string `json:"state"`
	Country string `json:"country"`
}

File: frontend/src/types/seller.ts
  // frontend/src/types/seller.ts
  export interface Seller {
    id: string;
    nickname: string;
    reputation: SellerReputation;
    tags: string[];
    location: Location;
    experience_years: number;
    leader_status: string;
  }
  
  export interface SellerReputation {
    level: number;
    status: string;
    transactions: TransactionMetrics;
    ratings: RatingMetrics;
  }
  
  export interface TransactionMetrics {
    completed: number;
    canceled: number;
    total: number;
  }
  
  export interface RatingMetrics {
    positive: number;
    neutral: number;
    negative: number;
  }
  
  export interface Location {
    city: string;
    state: string;
    country: string;
  }
  
  export interface Review {
    id: string;
    product_id: string;
    user_id: string;
    username: string;
    rating: number;
    comment: string;
    helpful_count: number;
    created_at: string;
  }
  
  export interface Question {
    id: string;
    product_id: string;
    question: string;
    answer?: string;
    created_at: string;
    answered_at?: string;
  }
  